### PR TITLE
Configurable build resources - integration

### PR DESCRIPTION
**Pull request by magicmatatjahu**
**Description**

Changes proposed in this pull request:

Configurable build resources - integration:
- update image for `function-ctrl` and `function-webhook` and configMap with envs for webhook
- update image for `console-web` and `console-backend`
- update documentation with information about presets
- update script for adjust runtime field to adjust `buildResources` field

**Related issue(s)**
https://github.com/kyma-project/kyma/issues/9227

### CODE REVIEW

File: docs/serverless/05-02-serverless-chart-config.md [hunk 1]
```diff
@@ -18,9 +18,15 @@ This table lists the configurable parameters, their descriptions, and default va
...
-| **containers.manager.envs.buildRequestsCPU**    | Number of CPUs requested by the image-building Pod to operate.        | `700m`        | `100m`            |
-| **containers.manager.envs.buildRequestsMemory** | Amount of memory requested by the image-building Pod to operate.      | `700Mi`       | `200Mi`           |
-| **containers.manager.envs.buildLimitsCPU**      | Maximum number of CPUs available for the image-building Pod to use.   | `1100m`       | `200m`            |
-| **containers.manager.envs.buildLimitsMemory**   | Maximum amount of memory available for the image-building Pod to use. | `1100Mi`      | `400Mi`           |
+| **webhook.values.buildJob.resources.minRequestCpu**    | Minimum number of CPUs requested by the image-building Pod to operate.        | `200m`        | `200m`   |
+| **webhook.values.buildJob.resources.minRequestMemory** | Minimum amount of memory requested by the image-building Pod to operate.      | `200Mi`       | `200Mi`  |
+| **webhook.values.buildJob.resources.defaultPreset**    | Default *preset for image-building Pod's resources.      | `normal`        | `local-dev`   |
```
Comment (kazydek) [inline docs/serverless/05-02-serverless-chart-config.md:23]:
```suggestion
| **webhook.values.buildJob.resources.defaultPreset**    | Default preset for image-building Pod's resources.      | `normal`        | `local-dev`   |
```

File: docs/serverless/05-02-serverless-chart-config.md [hunk 2]
```diff
@@ -18,9 +18,15 @@ This table lists the configurable parameters, their descriptions, and default va
...
+| **webhook.values.function.replicas.minValue**      | Minimum number of replicas of single Function.   | `1`       | `1`            |
```
Comment (kazydek) [inline docs/serverless/05-02-serverless-chart-config.md:24]:
```suggestion
| **webhook.values.function.replicas.minValue**      | Minimum number of replicas of a single Function.   | `1`       | `1`            |
```

File: docs/serverless/05-02-serverless-chart-config.md [hunk 3]
```diff
@@ -18,9 +18,15 @@ This table lists the configurable parameters, their descriptions, and default va
...
+| **webhook.values.function.replicas.defaultPreset**      | Default *preset for Function's replicas.   | `S`       | `S`            |
```
Comment (kazydek) [inline docs/serverless/05-02-serverless-chart-config.md:25]:
```suggestion
| **webhook.values.function.replicas.defaultPreset**      | Default preset for Function's replicas.   | `S`       | `S`            |
```

File: docs/serverless/05-02-serverless-chart-config.md [hunk 4]
```diff
@@ -18,9 +18,15 @@ This table lists the configurable parameters, their descriptions, and default va
...
+| **webhook.values.function.resources.minRequestCpu**      | Maximum number of CPUs available for the image-building Pod to use.   | `10m`       | `10m`            |
+| **webhook.values.function.resources.minRequestMemory**   | Maximum amount of memory available for the image-building Pod to use. | `16Mi`      | `16Mi`           |
+| **webhook.values.function.resources.defaultPreset**      | Default *preset for Function's resources.   | `M`       | `M`            |
```
Comment (kazydek) [inline docs/serverless/05-02-serverless-chart-config.md:28]:
```suggestion
| **webhook.values.function.resources.defaultPreset**      | Default preset for Function's resources.   | `M`       | `M`            |
```

File: docs/serverless/05-02-serverless-chart-config.md [hunk 5]
```diff
@@ -18,9 +18,15 @@ This table lists the configurable parameters, their descriptions, and default va
...
+
+*preset - predefined group of values. Read [available presets](#details-supported-webhooks-available-presets) to find out more.
```
Comment (kazydek) [inline docs/serverless/05-02-serverless-chart-config.md:30]:
```suggestion

```

File: docs/serverless/06-01-function-cr.md [hunk 6]
```diff
@@ -102,6 +110,10 @@ This table lists all the possible parameters of a given resource together with t
 | **spec.resources.limits.memory**         |       No       | Defines the maximum amount of memory available for the Function's Pod to use.      |
 | **spec.resources.requests.cpu**          |       No       | Specifies the number of CPUs requested by the Function's Pod to operate.       |
 | **spec.resources.requests.memory**       |       No       | Specifies the amount of memory requested by the Function's Pod to operate.               |
+| **spec.buildResources.limits.cpu**            |       No       | Defines the maximum number of CPUs available for the Kubernetes Job's Pod responsible for building Function's image to use.      |
```
[the remark this answers is not in the document]
Comment (kazydek) [reply docs/serverless/06-01-function-cr.md:113]: I changed the order as it looks way too long and unclear in this order when you replace `Function's Pod` with `Kubernetes Job's Pod responsible for building the Function's image`.

File: docs/serverless/03-03-supported-webhooks.md [hunk 7]
```diff
@@ -9,31 +9,44 @@ A newly created or modified Function CR is first updated by the defaulting webho
...
+> **NOTE:** Function's resources and replicas as well as resources for a Kubernetes Job are based on presets. Read about all [available presets](#details-available-presets) to find out more.
 
 ## Validation webhook
 
 It checks the following conditions for these CRs:
 
 1. [Function CR](#custom-resource-function)
 
-   - Minimum values requested for CPU, memory, and replicas are not lower than the required ones:
+   - Minimum values requested for CPU, memory, and replicas for a Function and CPU as well as memory for a Kubernetes Job responsible for building the Function's image must not be lower than the required ones:
```
Comment (kazydek) [inline docs/serverless/03-03-supported-webhooks.md:41]:
```suggestion
   - Minimum values requested for a Function (CPU, memory, and replicas) and a Kubernetes Job (CPU and memory) responsible for building the Function's image must not be lower than the required ones:
```

**Review by kazydek — DISMISSED** [review]

**Review by aerfio — DISMISSED** [review]

**Review by Wawrzyn321 — DISMISSED** [review]

**Review by kazydek — DISMISSED** [review]

**Review by aerfio — APPROVED** [review]

**Review by kazydek — APPROVED** [review]

**Review by kwiatekus — APPROVED** [review]

**Review by piotrmsc — APPROVED** [review]

**Pull request merged**